### PR TITLE
Refactor backend for wcs regions

### LINKED ISSUE #334: Cannot import/export regions when image has invalid or non-existent WCS headers
Some images don't have WCS information. For example, the "LMCq.fits" file has a minimal header, shown below. As such, it has no WCS information. The backend should still be able to export regions defined for this file in _pixel_ coordinates (although it's totally fine for it to fail to export in WCS). Currently, the backend fails to export regions, and displays `Export carta region type 3 failed: AnnotationBase: Coordinate system has no direction coordinate`. 

LMCq.fits header:
```
NAXIS1 = 10482
CRVAL1 = 85.0539
CRPIX1 = 5241
CRVAL2 = -70.7087
NAXIS2 = 10448
NAXIS = 2
CRPIX2 = 5224
DATE-OBS = 2017-10-21T01:29:00
```

### LINKED ISSUE #445: WCS regions: remove Region code from Frame
Currently, a Region belongs to a Frame so all region-handling code is in the Frame.  With WCS ROI, a region may be applied to any Frame in a WCS group, so it needs to be independent of a particular Frame until a data stream is needed. This means the file loaders and statistics calculators must be available to both.

The plan (in progress and subject to change) is documented [here](https://docs.google.com/document/d/1aN2ex1OHZ7_GQNVbagJpDNXnjQn8jIFIyiuIfQ8kq1Y/edit#).

The new architecture will make merges difficult if not impossible.  I plan to wait until major current development items from others are merged into dev, then manually add the changes into the new organization.

### PR DESCRIPTION
**Pull request by pford**
Closes #445.  Builds on previous (closed without merge) PR #517 with additional workarounds for region import/export.

Should also:
- fix #334 (region import/export for image with no direction coordinate)
- improve #505 (performance loading DS9 regions)

Does not fix:
- #528 (import ICRS region to FK5 ), pending casa6 PR then carta-casacore submodule update
- regions for images with rotation/non-square pixels (will create new issue)

### CODE REVIEW

File: Frame.cc [hunk 1]
```diff
@@ -144,543 +135,94 @@ int Frame::CurrentStokes() {
...
-        } else {
-            auto region = std::unique_ptr<carta::Region>(
-                new carta::Region(name, type, points, rotation, _image_shape, _spectral_axis, _stokes_axis, coord_sys));
-            if (region->IsValid()) {
-                _regions[region_id] = move(region);
-                region_set = true;
-            }
-        }
+    // Normalize any constants
+    ChannelRange range(chan_range.from, chan_range.to);
```
Comment (confluence) [inline Frame.cc]: Very minor nitpick: if we're going to copy `ChannelRange`s like this, we should add a constructor with a single `ChannelRange` parameter.

**Review by confluence — APPROVED** [review]
It looks like there aren't any additional changes to the HDF5 code (beyond what was discussed in the previous PR) -- I don't have any other specific comments.

File: Frame.cc (hunk 1 above)
Comment (pford) [reply Frame.cc]: Changed to use int variables instead for image slicer.

**Review by kswang1029 — APPROVED** [review]
tested region import/export. 👍

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**